plugins/module_utils/network/sonic/config/bgp_communities/bgp_communities.py

```diff
@@ -140,34 +140,67 @@ def set_state(self, want, have):
         if state == 'overridden':
             commands, requests = self._state_overridden(want, have, diff)
         elif state == 'deleted':
-            commands, requests = self._state_deleted(want, have, diff)
+            commands, requests = self._state_deleted(want, have)
         elif state == 'merged':
             commands, requests = self._state_merged(want, have, diff)
         elif state == 'replaced':
             commands, requests = self._state_replaced(want, have, diff)
         return commands, requests
 
-    @staticmethod
-    def _state_replaced(**kwargs):
+    def _state_replaced(self, want, have, diff):
         """ The command generator when state is replaced
 
         :rtype: A list
         :returns: the commands necessary to migrate the current configuration
                   to the desired configuration
         """
         commands = []
-        return commands
+        requests = []
+        requests_del = []
+        requests_replace = []
+
+        if diff:
+            for cmd in diff:
+                name = cmd['name']
+                requests_del.append(self.get_delete_single_bgp_community_requests(name))
+
+            if len(requests_del) > 0:
+                commands.extend(update_states(diff, "deleted"))
+                requests.extend(requests_del)
+
+            requests_replace = self.get_modify_bgp_community_requests(want, have)
+            if len(requests_replace) > 0:
+                commands.extend(update_states(want, "replaced"))
+                requests.extend(requests_replace)
+
+        return commands, requests
```

Comment:
Replace of the configuration is performed only when there is new/modified configuration of existing configuration in 'want'. If 'want' is a subset of 'have', then the non-specified configurations for that community in 'have' are not deleted.
Please handle the above scenario.